src/commands/deploy.ts

```diff
@@ -120,7 +120,22 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         const warning: string = `Are you sure you want to deploy to "${node.treeItem.client.fullName}"? This will overwrite any previous deployment and cannot be undone.`;
         context.properties.cancelStep = 'confirmDestructiveDeployment';
         const deployButton: vscode.MessageItem = { title: 'Deploy' };
-        await ext.ui.showWarningMessage(warning, { modal: true }, deployButton, DialogResponses.cancel);
+        if (defaultWebAppToDeploy) {
+            const changeDefault: vscode.MessageItem = { title: 'Change default' };
+            const result: vscode.MessageItem = await ext.ui.showWarningMessage(warning, { modal: true }, deployButton, DialogResponses.cancel, changeDefault);
+            if (result.title === 'Change default') {
+                const localRootPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.path : "";
+                const settingsPath = path.join(localRootPath, '.vscode', 'settings.json');
+                const doc = await vscode.workspace.openTextDocument(vscode.Uri.file(settingsPath));
+                vscode.window.showTextDocument(doc);
+                const workspaceConfiguration: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(constants.extensionPrefix, vscode.Uri.file(fsPath));
+                await workspaceConfiguration.update(constants.configurationSettings.defaultWebAppToDeploy, '');
+                await vscode.commands.executeCommand('appService.Deploy');
```

Comment:
It's obvious to us now, but I think a comment here explaining why there's a recursive `appService.Deploy` call would be useful for future reference. :)